doc/developer/developing.rst

```diff
@@ -286,3 +286,49 @@ Once you have written your own syntax checker, why not `submit a pull request
 <https://github.com/flycheck/flycheck/pulls>`__ to integrate it into Flycheck?
 If it's useful to you, it may be useful for someone else!  Please do check out
 our :ref:`flycheck-contributors-guide` to learn how we deal with pull requests.
+
+Issues with auto-quoting in `flycheck-define-checker`
+-----------------------------------------------------
+
+You may have noticed that lists passed to the ``:command`` or
+``:error-patterns`` in the snippets above are not quoted.  That is because
+`flycheck-define-checker` is a macro which automatically quotes these arguments
+(not unlike `use-package`).
+
+While this makes for a less noisy syntax, it unfortunately prevents you from
+defining a checker with compile-time arguments.  For example, you may be tempted
+to have a custom checker in your Emacs configuration written like this:
+
+.. code-block:: elisp
+
+   (flycheck-define-checker my-foobar-checker
+     :command ("foobar" source)
+     :error-patterns ((error …))
+     :modes `(foobar-mode ,my-other-foobar-mode))
+
+The idea is that you know statically one mode that you want to use the checker
+in: ``foobar-mode``, but another mode can be given via the variable
+``my-other-foobar-mode`` before the checker is defined.  In theory, this should
+work, but in practice it won't, since the ``:modes`` property is auto-quoted by
+`flycheck-define-checker`.  The issue arise not just with ``:modes``:, but with
+almost all the other properties since they are also auto-quoted.
```

Comment:
```suggestion
``my-other-foobar-mode`` before the checker is defined.  This won't work,
because the ``:modes`` property is auto-quoted by `flycheck-define-checker`.
The issue arise not just with ``:modes``:, but with almost all the other 
properties since they are also auto-quoted.
```